packages/components/src/utils/useShortcut.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useEffect, useLayoutEffect, useRef } from 'react';
+
+export const useShortcut = (key: string, callback: () => void) => {
+  const callbackRef = useRef(callback);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+
+  const handleKeyPress = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.ctrlKey && event.altKey && event.key === key) {
```

Comment:
what is when we want to add a shortcut with other modifiers?

Reply:
And as also said on the other PR, mac uses mostly cmd for short cuts and not crtl. Correct me if I'm wrong @ivy-lmu

Reply:
> what is when we want to add a shortcut with other modifiers?

I think we should not add a shortcut with other modifiers if at all possible. For consistency in the way our custom shortcuts are accessed and because this is the least used modifier combination by browsers. However, I am fine with adding an optional `options` argument to change the modifiers but keeping this as the default just in case.

Reply:
Not sure about this. E.g. Yesterday we talked about the idea if you are in the new dialog and e.g. press enter with the ctrl key (or whatever) then a new thing gets created without leaving the dialog, so you can directly add an additional one.

Reply:
I'm not saing that the default should not be with ctrl + alt.

Reply:
e.g. glsp does something like the following:

```ts
export function matchesKeystroke(event: KeyboardEvent, code: KeyCode, ...modifiers: KeyboardModifier[]): boolean {
    if (getActualCode(event) !== code)
        return false;
    if (isMac()) {
        if (event.ctrlKey !== (modifiers.findIndex(m => m === 'ctrl') >= 0))
            return false;
        if (event.metaKey !== (modifiers.findIndex(m => m === 'meta' || m === 'ctrlCmd') >= 0))
            return false;
    } else {
        if (event.ctrlKey !== (modifiers.findIndex(m => m === 'ctrl' || m === 'ctrlCmd') >= 0))
            return false;
        if (event.metaKey !== (modifiers.findIndex(m => m === 'meta') >= 0))
            return false;
    }
    if (event.altKey !== (modifiers.findIndex(m => m === 'alt') >= 0))
        return false;
    if (event.shiftKey !== (modifiers.findIndex(m => m === 'shift') >= 0))
        return false;
    return true;
}

export function isMac(): boolean {
    return window.navigator.userAgent.indexOf("Mac") !== -1;
}
```

at least as hint how to do this cmd or ctrl button ;)